src/execution/BucketExecutor.tsx

```diff
@@ -0,0 +1,338 @@
+// (C) 2007-2018 GoodData Corporation
+// tslint:disable:member-ordering
+import * as React from "react";
+import { SDK, factory as createSdk } from "@gooddata/gooddata-js";
+import { AFM, Execution, VisualizationObject, VisualizationInput } from "@gooddata/typings";
+
+import { setTelemetryHeaders } from "../helpers/utils";
+import { convertErrors } from "../helpers/errorHandlers";
+import { RuntimeError } from "../errors/RuntimeError";
+import { convertBucketsToAFM } from "../helpers/conversion";
+import { MEASURES, ATTRIBUTES } from "../constants/bucketNames";
+import { MEASUREGROUP } from "../internal/constants/bucket";
+
+type dimensionItem = VisualizationInput.AttributeOrMeasure | VisualizationInput.IAttribute;
+
+export const getExecutionFromDimensions = (
```

Comment:
This function is quite complex, yet there are no direct tests for it. Are you sure the rendering tests cover all the cases?

Reply:
added